be/test/runtime/snapshot_loader_test.cpp

```diff
@@ -339,5 +380,115 @@ TEST_F(SnapshotLoaderTest, DirMoveTaskIsIdempotent) {
+    }
+
+    int64_t dest_tablet_id = 333;
+    int32_t dest_schema_hash = 444;
+    std::string dest_path = fmt::format("{}/dest_snapshot/{}/{}", storage_root_path, dest_tablet_id,
+                                        dest_schema_hash);
+    ASSERT_TRUE(io::global_local_filesystem()->create_directory(dest_path).ok());
+
+    std::string src_hdr = remote_tablet_path + "/" + std::to_string(tablet_id) + ".hdr";
+    std::string dst_hdr = remote_tablet_path + "/" + std::to_string(dest_tablet_id) + ".hdr";
+    ASSERT_TRUE(io::global_local_filesystem()->rename(src_hdr, dst_hdr).ok());
+    auto guards = engine_ref->snapshot_mgr()->convert_rowset_ids(
+            remote_tablet_path, dest_tablet_id, 0, 0, partition_id, dest_schema_hash);
+
+    // 7. Setup a remote tablet snapshot for download
+    TRemoteTabletSnapshot remote_snapshot;
+    remote_snapshot.remote_tablet_id = dest_tablet_id;
+    remote_snapshot.local_tablet_id = tablet_id;
+    remote_snapshot.local_snapshot_path = snapshot_path;
+    remote_snapshot.remote_snapshot_path = remote_tablet_path;
+    remote_snapshot.remote_be_addr.hostname = "127.0.0.1";
+    remote_snapshot.remote_be_addr.port = 1234;
+    remote_snapshot.remote_token = "fake_token";
+
+    // 8. Download the snapshot
+    std::vector<TRemoteTabletSnapshot> remote_snapshots = {remote_snapshot};
+    std::vector<int64_t> downloaded_tablet_ids;
+    SnapshotLoader loader(*engine_ref, ExecEnv::GetInstance(), 3L, tablet_id);
+    status = loader.remote_http_download(remote_snapshots, &downloaded_tablet_ids);
+    ASSERT_TRUE(status.ok());
+
+    // 9. Verify file linking - check if files were properly linked rather than downloaded
+
+    // Find rowset data files in the download directory
+    std::vector<io::FileInfo> downloaded_files;
+    bool exists = false;
+    ASSERT_TRUE(io::global_local_filesystem()
+                        ->list(snapshot_path, true, &downloaded_files, &exists)
+                        .ok());
+    ASSERT_TRUE(exists);
+
+    // Look for linked files (files containing the new rowset ID that should be linked)
+    bool found_linked_file = false;
+    for (const auto& file : downloaded_files) {
+        if (file.file_name.find(".dat") != std::string::npos) {
+            found_linked_file = true;
+
+            // This is our linked file - verify it exists and has proper size
+            struct stat file_stat;
+            std::string file_path = snapshot_path + "/" + file.file_name;
+            int ret = stat(file_path.c_str(), &file_stat);
+            ASSERT_EQ(ret, 0) << "Failed to stat file: " << file_path;
+
+            // Verify file size matches expected data file size
+            ASSERT_GT(file_stat.st_size, 0) << "Linked file is empty: " << file_path;
+
+            std::cout << "Found linked file: " << file_path << ", size: " << file_stat.st_size
+                      << " bytes" << std::endl;
+            break;
+        }
+    }
```

Comment:
The goal here is to verify whether the file download step is skipped if the remote file has a resource_rowset_id that points to the local rowsets.